patches/minecraft/net/minecraft/world/gen/ChunkProviderServer.java.patch

```diff
@@ -1,14 +1,39 @@
 --- ../src-base/minecraft/net/minecraft/world/gen/ChunkProviderServer.java
 +++ ../src-work/minecraft/net/minecraft/world/gen/ChunkProviderServer.java
-@@ -35,6 +35,7 @@
+@@ -33,8 +33,31 @@
+     private final Set<Long> field_73248_b = Sets.<Long>newHashSet();
+     public final IChunkGenerator field_186029_c;
      public final IChunkLoader field_73247_e;
-     public final Long2ObjectMap<Chunk> field_73244_f = new Long2ObjectOpenHashMap(8192);
+-    public final Long2ObjectMap<Chunk> field_73244_f = new Long2ObjectOpenHashMap(8192);
++    // Forge start - optimize chunk lookups by using last access cache
++    public final Long2ObjectMap<Chunk> field_73244_f = new Long2ObjectOpenHashMap<Chunk>(8192) {
```

Comment:
1) Don't do this as a non class in a patch file, just makes it that much more of a pain when the patch fails to apply.
2) Doing it as a util class allows for it to be reused in other places if needed.
3) I understand the concept of this, but there needs to be SOME actual documented justification for this.